### PR TITLE
[FIX] LeftMenu collection type name sorting

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Sorts collection type names in the CM the same way as the CTB (see `packages/core/content-type-builder/admin/src/components/DataManagerProvider/utils/cleanData.js`)

image

### Why is it needed?

The order of content types in Content Manager was not alphabetical. It puts special letters like Č, Š, ... at the end. On the other hand, Content type builder have correct order.

### Related issue(s)/PR(s)

Fixes https://github.com/strapi/strapi/issues/15342

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js [hunk 1]
```diff
@@ -8,23 +8,27 @@ import React, { useMemo, useState } from 'react';
 import { useSelector, shallowEqual } from 'react-redux';
 import { useIntl } from 'react-intl';
 import matchSorter from 'match-sorter';
+import { NavLink } from 'react-router-dom';
```
Comment (gu-stav) [inline packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js:11]: Is that a mistake?

File: packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js [hunk 2]
```diff
@@ -8,23 +8,27 @@ import React, { useMemo, useState } from 'react';
...
   SubNavSections,
   SubNavLink,
 } from '@strapi/design-system/v2/SubNav';
+
 import getTrad from '../../../utils/getTrad';
 import { makeSelectModelLinks } from '../selectors';
 
 const matchByTitle = (links, search) =>
   search
     ? matchSorter(links, toLower(search), { keys: [(item) => toLower(item.title)] })
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js:30]: can we swap out `toLower` for the native impl `toLowerCase`? 🤔 
Feels unnecessary using lodash so much.

File: packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js [hunk 3]
```diff
@@ -8,23 +8,27 @@ import React, { useMemo, useState } from 'react';
...
-    : sortBy(links, (object) => object.title.toLowerCase());
+    : sortBy(links, (object) => camelCase(object.title));
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js:31]: Ideally i'd love to get rid of `sortBy` here too, if it's just an array of `object` we can probably write our own? wdyt @jhoward1994?

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Few Q's nothing major – good job 💪🏼

File: packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js (hunk 1 above)
Comment (jhoward1994) [reply packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js:11]: No, I was just reordering the imports in a way that made sense to me. I can move it back if needed?

**Review by joshuaellis — APPROVED** [review]
NICE ⭐ Great job getting rid of as much lodash as possible 💪🏼

**Pull request merged**